lib/utils/makeRequest.js

```diff
@@ -11,11 +11,14 @@ function xmlHttpRequest() {
   return new request();
 }
 
-function makeRequest(provider, token) {
+function makeRequest(data) {
```

Comment:
It  should probably be `makeRequest(provider, tokenOrUsername, password)`, where last two args are also optional.
If `2` args are passed it counts for sandbox token, if `3` for remote auth credentials.

Reply:
inside this function, should the local variable remain `token`, or should be renamed as `this.tokenOrUsername` ?

Reply:
It should remain  as `this.token` and difference will be only constructor's args. Constructor simply decides which `this` params should be filled.
I like your solution more and it's how it should be made to work in v1.0.0. For now just compromising in case anyone uses this `_makeRequest`.